src/components/themed/CarouselDots.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react'
+import { Pressable, View, type ViewStyle } from 'react-native'
+import Animated, {
+  type SharedValue,
+  useAnimatedStyle,
+  withTiming
+} from 'react-native-reanimated'
+
+import { cacheStyles, type Theme, useTheme } from '../services/ThemeContext'
+
+interface Props {
+  itemCount: number
+  scrollIndex: SharedValue<number>
+  onPress?: (index: number) => void
+}
+
+/**
+ * Renders the pagination dots at the bottom of a carousel.
+ * Tapping on a dot scrolls the carousel to the right item.
+ */
+export const CarouselDots: React.FC<Props> = props => {
+  const { itemCount, scrollIndex, onPress } = props
+  const theme = useTheme()
+  const styles = getStyles(theme)
+
+  return (
+    <View style={styles.dotRow}>
+      {Array.from({ length: itemCount }).map((_, index) => {
+        const handlePress = (): void => {
+          scrollIndex.value = withTiming(index)
+          if (onPress != null) onPress(index)
+        }
+        return (
+          <Pressable key={index} hitSlop={theme.rem(0.2)} onPress={handlePress}>
```

Comment:
Can further increase the `hitSlop` so there's no deadzones between the dots

Reply:
I tried, but there's a problem where the previous dot becomes un-hittable. This is as wide as I could get it without breaking.